theano/configdefaults.py

```diff
@@ -1681,6 +1681,18 @@ def default_compiledir():
         allow_override=False),
     in_c_key=False)
 
+
+# To resolve https://github.com/Theano/Theano/issues/3029.
+AddConfigVar(
+    'deterministic',
+    ("Avoids the use of the AtomicAdd operation which fails to guarantee the"
+     "deterministic properties of the addition because of the behavior of"
+     "floating-point arithmetic."
+     "This slows down execution by a large factor."),
```

Comment:
We should also say that it can trigger crashes if some non-deterministic operations are still in the graph.